ush/python/pygfs/task/aero_analysis.py

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python3
+
+import os
+import glob
+import gzip
+import tarfile
+from netCDF4 import Dataset
+from logging import getLogger
+from typing import Dict, List, Any
+
+from pygw.attrdict import AttrDict
+from pygw.file_utils import FileHandler
+from pygw.timetools import to_isotime, to_fv3time, to_timedelta
+from pygw.fsutils import rm_p
+from pygw.template import Template, TemplateConstants
+from pygw.yaml_file import YAMLFile
+from pygw.logger import logit
+from pygfs.task.analysis import Analysis
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class AerosolAnalysis(Analysis):
+    """
+    Class for global aerosol analysis tasks
+    """
+    @logit(logger, name="AerosolAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+
+        _res = int(self.config['CASE'][1:])
+        _res_enkf = int(self.config['CASE_ENKF'][1:])
+        _window_begin = self.runtime_config.current_cycle - to_timedelta(f"{self.config['assim_freq']}H") / 2
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_enkf + 1,
+                'npy_anl': _res_enkf + 1,
+                'npz_anl': self.config['LEVS'] - 1,
+                'AERO_WINDOW_BEGIN': to_isotime(_window_begin),
+                'AERO_WINDOW_LENGTH': f"PT{self.config['assim_freq']}H",
+                'BKG_ISOTIME': to_isotime(self.runtime_config.current_cycle),
+                'BKG_YYYYmmddHHMMSS': to_fv3time(self.runtime_config.current_cycle),
+                'cdate_fv3': to_fv3time(self.runtime_config.current_cycle),
+                'comin_ges_atm': self.config.COMIN_GES.replace('chem', 'atmos'),  # 'chem' is COMPONENT, aerosol fields are in 'atmos' tracers
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
+        """Initialize a global aerosol analysis
+
+        This method will initialize a global aerosol analysis using JEDI.
+        This includes:
+        - staging CRTM fix files
+        - staging FV3-JEDI fix files
+        - staging B error files
+        - staging model backgrounds
+        - generating a YAML file for the JEDI executable
+        - linking the JEDI executable (TODO make it copyable, requires JEDI fix)
+        - creating output directories
+        """
+        super().initialize()
```

Comment:
same as above, but that line doesn't seem to do anything currently. Is there a future plan for something to be done in analysis's initialize?

Reply:
I think the intention is once we have atm, soca, land, etc. we can find where things are common and put them in the parent method.